provisor/provisor.py

```diff
@@ -9,6 +9,11 @@
 from utils import make_salt, drop_privileges
 from random import shuffle
 
+# An exception class for erroring-out
+#  on unknown hosts
+class UNKNOWN_HOST(Exception):
```

Comment:
If we want to implement more kinds of errors, we probably ought to create a superclass for provisor's errors (and give it a method to return a `{"message": foo}` dictionary?)